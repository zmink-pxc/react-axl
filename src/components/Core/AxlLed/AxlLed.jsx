import React from 'react';
import classNames from 'classnames';
import styles from '@bit/zmink.axioline.axl-ledmodule';

export default class AxlLed extends React.PureComponent {
    constructor(props){
        super(props)
    }

    render(){
        const ledClass = classNames(styles.led_base,{[styles.error]:this.props.isError},{[styles.warning]:this.props.isWarning},{[styles.active]:this.props.isActive})
        return (<div className={ledClass}>
            </div>)
        
    }
}

//isError
//isWarning
//isActive