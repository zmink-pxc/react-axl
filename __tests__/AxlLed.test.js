import React from 'react';
import AxlLed from '../src/components/Core/AxlLed/AxlLed';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  var component = renderer.create(<AxlLed/>)
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  component = renderer.create(<AxlLed isError={true}/>)
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  component = renderer.create(<AxlLed isActive={true}/>)
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  component = renderer.create(<AxlLed isWarning={true}/>)
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});