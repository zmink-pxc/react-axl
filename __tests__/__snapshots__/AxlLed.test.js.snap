// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="led_base"
/>
`;

exports[`renders correctly 2`] = `
<div
  className="led_base error"
/>
`;

exports[`renders correctly 3`] = `
<div
  className="led_base active"
/>
`;

exports[`renders correctly 4`] = `
<div
  className="led_base warning"
/>
`;
